### PR TITLE
planning checklist blog

### PR DESCRIPTION
**Pull request by ankeetmaini**

### CODE REVIEW

File: content/posts/planning-checklist.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+so with above context, how do you decide what gets done in a quarter? what actually is **planning**?
+
+> you can wing it yolo style because life's too short to plan anything, because seriously who cares?
+
+**actually you should, as if your life depends on it**
+
+i've done it both ways (just don't tell my boss though), planned (roadmaps, sprints) on the fly and derived them meticulously by using the below framework. the difference was monumental in the quality of deliverables and the morale of the team!
+
+earlier i'd have a meeting every two weeks, rethinking priorities and not just me, my entire team. i thought having them on the call would give them visibility but all they saw was uncertainty and jitteriness. they felt as if nobody had any idea what exactly needed to be done, and everyone was busy pitching things they would like to do instead which might not be in the best interest of the organisation. like everyone can't work on reducing build times or trying out a new framework which came out yesterday!
```
Comment (siddharthkp) [inline content/posts/planning-checklist.md:29]: optional: the nobody and everyone confused me, maybe more words for some clarity 

```suggestion
earlier i'd have a meeting every two weeks, rethinking priorities and not just me, my entire team. i thought having them on the call would give them visibility but all they saw was uncertainty and jitteriness. they felt as if nobody had any idea what exactly needed to be done, and the entire team was busy pitching things they would like to do instead which might not be in the best interest of the organisation. doesn’t make sense for everyone to work on reducing build times or trying out a new framework which came out yesterday!
```

File: content/posts/planning-checklist.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+so i had to change, and below is what i put together after i got humiliated at a party with my peers and bosses :P (no surprises there)
```
Comment (siddharthkp) [inline content/posts/planning-checklist.md:31]: optional: gossip! share more details?

File: content/posts/planning-checklist.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+</tbody>
+</table>
+
+<div class="message-box">
+
+e.g. 
+- objective: improve the conversion funnel by 20% on our e-commerce app
+- key results:
+  - reduce page load of product page from 6s → 3.5s
+  - remove friction on payment page by showing up upi as the top payment mode
```
Comment (siddharthkp) [inline content/posts/planning-checklist.md:57]:
```suggestion
  - remove friction on payment page by showing UPI as the top payment mode
```

**Review by siddharthkp — COMMENTED** [review]
i only reviewed half of the article for now, posting the comments early

treat everything as optional, of course

File: content/posts/planning-checklist.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+layout: layouts/post.njk
+date: 2024-05-13
+tags:
+  - leadership
+  - planning
+---
+
+it's almost may and in just about a month a brand new quarter would start which comes with a possibility of endless opportunities in engg roadmap parlance. 
+
+we've come a long way from knowing what would get done years in advance, like a construction building, say a new mall for which once the blueprint is finalised, you can usually estimate the multi-year work upfront at least on paper before you dive deep into execution. but who knew building software is even tougher or have we deliberately made it that way?
```
[the remark this answers is not in the document]
Comment (ankeetmaini) [reply content/posts/planning-checklist.md:13]: +1, although i was thinking that the fuzziness of it makes it even more complex. you set out to build one thing but then something else gets built as the ideas evolve

File: content/posts/planning-checklist.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+we work in sprints, we put our heads down for two weeks and then again and again. some teams decide within these two weeks what next to do and sort of groom their stories with no hints or indication of what lies ahead and coins words like agile :P
+
+but there's just too much flux in this way. there's so much extra time being spent deciding every two weeks what to do. does it even matter right now or can be pushed out?
+
+in real world, work happens around big projects and launches. and they have dates or an expected date of launch. everything begins from there and you usually back calculate.
```
[the remark this answers is not in the document]
Comment (ankeetmaini) [reply content/posts/planning-checklist.md:19]: reworded it to companies.

File: content/posts/planning-checklist.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+so with above context, how do you decide what gets done in a quarter? what actually is **planning**?
+
+> you can wing it yolo style because life's too short to plan anything, because seriously who cares?
+
+**actually you should, as if your life depends on it**
```
[the remark this answers is not in the document]
Comment (ankeetmaini) [reply content/posts/planning-checklist.md:25]: ok, i see. i’m saying you should plan meticulously.

[the remark this answers is not in the document]
Comment (ankeetmaini) [reply content/posts/planning-checklist.md]: see if it reads better?

File: content/posts/planning-checklist.md (hunk 2 above)
Comment (ankeetmaini) [reply content/posts/planning-checklist.md:31]: i actually started bragging how i do sprint plannings and then everyone was like; dude that’s the worst way to do it. lol

**Review by bhaveshraheja — COMMENTED** [review]
LGTM

A suggestion: add like a takeaways section towards the end with some points. You kind of already did with the "hire smart people and not telling them how to do stuff"

File: content/posts/planning-checklist.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+so with above context, how do you decide what gets done in a quarter? what actually is **planning**?
+
+> you can choose to wing it yolo style because life's too short to plan anything, because seriously who cares?
+
+**but you should care, as if your life depends on it**
+
+i've done it both ways (just don't tell my boss though), planned (roadmaps, sprints) on the fly and derived them meticulously by using the below framework. the difference was monumental in the quality of deliverables and the morale of the team!
+
+earlier i'd have a meeting every two weeks, rethinking priorities and not just me, my entire team. i thought having them on the call would give them visibility but all they saw was uncertainty and jitteriness. they felt as if nobody had any idea what exactly needed to be done, and the entire team was busy pitching things they would like to do instead which might not be in the best interest of the organisation. it doesn’t make sense for everyone to work on reducing build times or trying out a new framework which came out yesterday!
```
[the remark this answers is not in the document]
Comment (ankeetmaini) [reply content/posts/planning-checklist.md:29]: so on point!

**Review by ankeetmaini — COMMENTED** [review]
fix a comment

File: content/posts/planning-checklist.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+- key result: reduce page load of product page from 6s → 3.5s
+- initiatives
+  - reduce the javascript bundle size by 120kb to improve time to interactive
+    - remove lodash
+    - write a lighter implementation for the product page image component which is 100kb in size
+  - 🔺 enable edge caching for product listing call to speed up the api response time
+  - 🔺 partial painting of product page reusing the state of homepage while the api call is in progress
+
+> 🔺 means that the initiative needs refinement and is a fuzzy item. it will usually require an hld/lld/1-pager and would probably get executed as a spike item where concrete details would get added after a deep-dive or a small poc.
+>> the staff+ engineers would also have to make a case as to why do they think the above initiative is needed with roi in mind as well, because such fuzzy initiatives require greater engineering efforts and often run into risk during execution
```
[the remark this answers is not in the document]
Comment (ankeetmaini) [reply content/posts/planning-checklist.md:94]: _on the off chance, if all of your items are fuzzy, then it means that the planning is broken and the team is just putting all eggs into a singe basket, basket of uncertainty. in that case, adjust your goals to collect more data/insights first before jumping to solve the problem_

how does this look?

**Pull request merged**